src/libs/ajax.js

```diff
@@ -52,6 +52,23 @@ const withCancellation = wrappedFetch => async (...args) => {
   }
 }
 
+const withRetryOnError = _.curry(wrappedFetch => async (...args) => {
+  const timeout = 5000
+  const somePointInTheFuture = Date.now() + timeout
+  const maxDelayIncrement = 1500
+  const minDelay = 500
+
+  while (Date.now() < somePointInTheFuture) {
+    const until = Math.random() * maxDelayIncrement + minDelay
```

Comment:
What was the thinking behind the random delay? It seems unnecessarily complex to me.

Reply:
@slucasbroad can you provide the background on why you are doing a random delay?

Reply:
Using Googles recommend retry algorithm. If a lot of clients are retrying the same service, random prevents them from retrying at the same time, staggering the requests.